### ISSUE TITLE
conda install from tarball error?

### ISSUE THREAD

**Issue by jakirkham** [conda/conda#2886]
Running into this issue when trying to install directly from a tarball.

```
Traceback (most recent call last):
  File "/usr/local/bin/conda2", line 6, in <module>
    sys.exit(main())
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


  File "/opt/conda2/lib/python2.7/site-packages/conda/cli/main.py", line 120, in main
    exit_code = args_func(args, p)
  File "/opt/conda2/lib/python2.7/site-packages/conda/cli/main.py", line 130, in args_func
    exit_code = args.func(args, p)
  File "/opt/conda2/lib/python2.7/site-packages/conda/cli/main_install.py", line 69, in execute
    install(args, parser, 'install')
  File "/opt/conda2/lib/python2.7/site-packages/conda/cli/install.py", line 196, in install
    explicit(args.packages, prefix, verbose=not args.quiet)
  File "/opt/conda2/lib/python2.7/site-packages/conda/misc.py", line 79, in explicit
    is_file = schannel.startswith('file:') and schannel.endswith('/')
UnboundLocalError: local variable 'schannel' referenced before assignment
```

**Comment by kalefranz** [comment]
`conda info`?

**Comment by jakirkham** [comment]
This is in a docker image.

```
$ conda info
Current conda install:

             platform : linux-64
        conda version : 4.1.4
    conda-env version : 2.5.1
  conda-build version : 1.20.0
       python version : 2.7.11.final.0
     requests version : 2.9.2
     root environment : /opt/conda2  (writable)
  default environment : /opt/conda2
     envs directories : /opt/conda2/envs
        package cache : /opt/conda2/pkgs
         channel URLs : https://conda.anaconda.org/nanshe/linux-64/
                        https://conda.anaconda.org/nanshe/noarch/
                        https://conda.anaconda.org/conda-forge/linux-64/
                        https://conda.anaconda.org/conda-forge/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /root/.condarc
         offline mode : False
    is foreign system : False
```

**Comment by mcg1969** [comment]
I'm having trouble reproducing this.

**Comment by mcg1969** [comment]
I know this should be obvious but can you give us the exact command line?

**Comment by mcg1969** [comment]
@jakirkham More details here would be helpful, including the specific package you tried to install by tarball. I'm genuinely having difficulty reproducing.

**Comment by jakirkham** [comment]
So, this is coming out of an open sourced Docker image. Though I really should come up with a simpler example. I started to and then other stuff came up. I'll give it another try.

Until I do here is the [Dockerfile]( https://github.com/nanshe-org/docker_nanshe/blob/6fa60ad6f221731c17bf2277f5744f8b781095db/Dockerfile ). Sorry it is so ugly. I've tried my best to make it readable given the constraints. The line that cause it to fail is this [one]( https://github.com/nanshe-org/docker_nanshe/blob/6fa60ad6f221731c17bf2277f5744f8b781095db/Dockerfile#L29 ).

Basically, what happens is we install everything in the `root` environment of two different `conda`s. One for Python 2 and the other for Python 3. We then remove one package `nanshe` and download the source code matching that version so as to have the test suite. We then remove the package and run the test suite. Once complete we try to reinstall the package from a file which fails.

**Comment by mcg1969** [comment]
I was able to reproduce the problem---it is specifically limited to installing tarballs _in the package cache_. As a short-term fix you can copy the package out of the cache and then reinstall, I think. #2907 is the same issue. I'm working on a fix now. cc: @kalefranz

**Comment by mcg1969** [comment]
OK. Got it. #2908 should take care of it. I've added two additional integration tests to exercise the case where tarballs are being installed from the package cache.

**Comment by mcg1969** [comment]
The nice thing about installing directly from the package cache is you get the channel metadata along with it.

**Comment by jakirkham** [comment]
Thanks for fixing this.

> The nice thing about installing directly from the package cache is you get the channel metadata along with it.

Definitely. That's why I like doing it. 😉
